model-server/src/main/kotlin/org/modelix/model/server/handlers/ModelReplicationServer.kt

```diff
@@ -239,17 +240,29 @@ class ModelReplicationServer(
 
     override suspend fun PipelineContext<Unit, ApplicationCall>.postRepositoryObjectsGetAll(repository: String) {
         checkPermission(ModelServerPermissionSchema.repository(repository).objects.read)
-        runWithRepository(repository) {
-            val keys = call.receiveStream().bufferedReader().use { reader ->
-                reader.lineSequence().toHashSet()
-            }
-            val objects = withContext(Dispatchers.IO) { modelClient.store.getAll(keys) }.checkValuesNotNull {
-                "Object not found: $it"
+        val channel = call.receiveChannel()
+        val keys = hashSetOf<String>()
+        while (true) {
+            val line = channel.readUTF8Line() ?: break
```

Comment:
This pattern seems to repeat below. What about factoring out something like `toLineStream`?

Reply:
Can't we go to a `Sequence` and then reuse the old implementation with `windowed`, which is overloaded for Sequence to return a Sequence.

Reply:
Not, sure if this is possible.
Sequences and Coroutines use suspending functions but in a different way.
![grafik](https://github.com/user-attachments/assets/02f33b40-bd16-4af0-8155-b5c59e4f6226)


> The sequence builder should not use suspending operations other than yielding operations[3](https://kt.academy/article/cc-sequence#definition-3). If you need, for instance, to fetch data, it's better to use Flow, as will be explained later in the book. The way its builder works is similar to the sequence builder, but Flow has support for other coroutine features.

See https://kt.academy/article/cc-sequence

Reply:
What a mess